src/runtime/particle-execution-context.ts

```diff
@@ -315,13 +328,18 @@ export class ParticleExecutionContext {
 
     let container = this.wasmContainers[spec.implFile];
     if (!container) {
-      const buffer = await this.loader.loadWasmBinary(spec);
-      if (!buffer || buffer.byteLength === 0) {
-        throw new Error(`Failed to load wasm binary '${spec.implFile}'`);
-      }
-
       container = new WasmContainer(this.loader, this.apiPort);
-      await container.initialize(buffer);
+
+      let initialized = false;
+      while (!initialized) {
```

Comment:
Looping on a fixed delay is not the best way to handle this:
- If someone loads a malformed binary, this will loop indefinitely.  Adding an arbitrary spin limit risks being too low to catch a slow compilation/load cycle and too high such that we still see spin locking effects.
- Depending on how the wasm module is being written to disk and how the instantiation process works, we might end up "successfully" loading a partially complete file.
- It adds unnecessary processing: while the binary is still being built, any attempt to load it is necessarily going to fail.
- Adding timed delays often complicates unit tests.

A better approach would be to build detection of the file write being completed into the file watching mechanism itself. Best case we can hook the CLOSE_WRITE inotify event (on Linux), but if the watcher simply delay loops to detect the end of file writes, that's still an improvement.